### PR TITLE
[pistache] Add new port

### PR DESCRIPTION
**Pull request by NancyLi1013**
This port currently only supports Linux platform.
Related issue #6732.

### CODE REVIEW

**Comment by NancyLi1013** [comment]
The failure on Mac OS is due to this issue https://github.com/oktal/pistache/issues/183.
And this port does not work on Mac OS now. See https://github.com/oktal/pistache/issues/1.

File: ports/pistache/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+vcpkg_configure_cmake(
+    SOURCE_PATH ${SOURCE_PATH}
+    PREFER_NINJA 
+)
+
+vcpkg_install_cmake()
+
+vcpkg_fixup_cmake_targets(CONFIG_PATH lib/cmake/${PORT})
+vcpkg_fixup_cmake_targets(CONFIG_PATH lib/pkgconfig)
```
Comment (dan-shaw) [inline ports/pistache/portfile.cmake]: `lib/pkgconfig` doesn't appear to contain any cmake targets. It might be better to remove this folder instead.

Comment (NancyLi1013) [reply ports/pistache/portfile.cmake]: Thanks for your remind. @dan-shaw
I have removed pkgconfig file.

**Comment by dan-shaw** [comment]
Thanks for the PR!

**Pull request merged**